onmt/trainer.py

```diff
@@ -269,15 +269,17 @@ def train(self,
                 step, train_steps,
                 self.optim.learning_rate(),
                 report_stats)
+            if step % self.report_manager.report_every == 0:
+                self._report_validation(self.optim.learning_rate(),
+                                        step, valid_stats=None)
 
             if (valid_iter is not None and step % valid_steps == 0 and
-                    self.gpu_rank == 0):
+                    self.gpu_rank <= 0):
 
                 valid_stats = self.validate(
                     valid_iter, moving_average=self.moving_average)
-                self._report_step(self.optim.learning_rate(),
-                                  step, train_stats=total_stats,
-                                  valid_stats=valid_stats)
+                self._report_validation(self.optim.learning_rate(),
+                                        step, valid_stats=valid_stats)
```

Comment:
I think it is still good to report "total_stats" especially the batch stats, the total examples processes, ....